src/accounts/context/userAccount.tsx

```diff
@@ -0,0 +1,97 @@
+// Libraries
+import React, {FC, useCallback, useEffect, useState} from 'react'
+// import {useParams, useHistory} from 'react-router-dom'
+import {useDispatch} from 'react-redux'
+
+// Types
+import {Account as UserAccount} from 'src/client/unityRoutes'
+
+// Utils
+import {getAccounts} from 'src/client/unityRoutes'
+
+export type Props = {
+  children: JSX.Element
+}
+
+export interface UserAccountContextType {
+  userAccounts: UserAccount[]
+  handleGetAccounts: () => void
+
+  defaultAccountId: number
+  activeAccountId: number
+}
+//   todo: add to above when implementing:
+//    setDefaultAccountId: (id: number) => void
+
+// isActive: true is for the currently logged in/active account
+export const DEFAULT_CONTEXT: UserAccountContextType = {
+  userAccounts: [],
+  defaultAccountId: -1,
+  activeAccountId: -1,
+  handleGetAccounts: () => {},
+}
+
+export const UserAccountContext = React.createContext<UserAccountContextType>(
+  DEFAULT_CONTEXT
+)
+
+// todo:  put in dependency array:  whenever the default account changes, should redo the call.
+// put this in *after* the ability to change the default account has been added
+
+export const UserAccountProvider: FC<Props> = React.memo(({children}) => {
+  const [userAccounts, setUserAccounts] = useState<UserAccount[]>(null)
+  const [defaultAccountId, setDefaultAccountId] = useState<number>(null)
+  const [activeAccountId, setActiveAccountId] = useState<number>(null)
+
+  const dispatch = useDispatch()
+
+  const handleGetAccounts = useCallback(async () => {
+    try {
+      const resp = await getAccounts({})
+      if (resp.status !== 200) {
+        // set user account status to error;...TODO
+        throw new Error(resp.data.message)
```

Comment:
almost universally, we're doing these kinds of requests in an `api/` layer so the view layer can be related to rendering and interacting with the user. Some examples for inspiration:

https://github.com/influxdata/ui/blob/master/src/organizations/apis/index.ts
https://github.com/influxdata/ui/blob/master/src/dashboards/apis/index.ts
https://github.com/influxdata/ui/blob/master/src/templates/api/index.ts

This being at this layer leads me to reiterate my previous question: why context instead of redux? We typically use context for front-end related constructs (like date display preferences) and redux for api-level data.

Reply:
i chose context because all of the other quartz components  (billing, operator account, etc) use contexts

if you really want me to change it, i can but i am following a pattern.